packages/edit-site/src/components/app/index.js

```diff
@@ -11,15 +11,30 @@ import { Routes } from '../routes';
 import Editor from '../editor';
 import List from '../list';
 import NavigationSidebar from '../navigation-sidebar';
+import HomepageRedirect from '../homepage-redirect';
 import getIsListPage from '../../utils/get-is-list-page';
 
+function getNeedsHomepageRedirect( params ) {
+	const { postType } = params;
+	return (
+		! getIsListPage( params ) &&
+		! [ 'post', 'page', 'wp_template', 'wp_template_part' ].includes(
+			postType
+		)
+	);
+}
+
 export default function EditSiteApp( { reboot } ) {
 	return (
 		<SlotFillProvider>
 			<UnsavedChangesWarning />
 
 			<Routes>
 				{ ( { params } ) => {
+					if ( getNeedsHomepageRedirect( params ) ) {
+						return <HomepageRedirect />;
```

Comment:
Maybe we should only render it in `/editor/index.js`? Since that we won't need to redirect if we're already on the list page.

Reply:
True, though rendering it here makes it easier to bypass all the react hooks in the `Editor` component.

Reply:
This is no longer an issue with the latest commit, as there's no longer a component 😄